src/bitmap.c

```diff
@@ -1118,6 +1118,7 @@ GdipCloneBitmapAreaI (INT x, INT y, INT width, INT height, PixelFormat format,
 	}
 
 	result->image_format = original->image_format;
+	result->active_bitmap->pixel_format = format;
```

Comment:
I am not sure if this is correct fix. The code below takes the `cairo_format` from the original bitmap (`result->cairo_format = original->cairo_format;` and it seems like it can result in inconsistent values.

Reply:
Hm, that seems possible. So the `result->cairo_format` should be updated too? To have a consistent state.

Or is there any other way to fix this behavior, which I'm not aware atm?

Reply:
It's more difficult than that I am afraid. At line 1185 the assumption about destination format == source format is still taken into account. I don't see any code doing the actual format conversion.

Reply:
Besides that, `PixelFormatDontCare` is not handled either. In that case it should use the source format.